test/datasets_utils.py

```diff
@@ -569,7 +573,13 @@ def create_image_file(
 
     image = create_image_or_video_tensor(size)
     file = pathlib.Path(root) / name
-    PIL.Image.fromarray(image.permute(2, 1, 0).numpy()).save(file, **kwargs)
+
+    # torch (num_channels x height x width) -> PIL (width x height x num_channels)
+    image = image.permute(2, 1, 0).numpy()
+    # For grayscale images PIL doesn't use a channel dimension
+    if image.shape[2] == 1:
+        image = np.squeeze(image, 2)
```

Comment:
nit: you can avoid having to import `numpy` if you do the squeeze while it's a tensor